MeasureAuthoringTool/bonnie
# Update Medications for Cumulative Medication Duration calculations

~~_Note: This feature requires the updated https://github.com/pophealth/hqmf2js/commit/6a7b24e54580d863f28fe90d99763932f57b8949 for accurate CMD calculations._~~
#### Story
- https://www.pivotaltracker.com/story/show/70168122
#### Notes
- removed [ DOSE, FREQUENCY, CUMULATIVE_MEDICATION_DURATION ] from Medication field values
- added Dose (dose_value, dose_unit) and Frequency (administrationTiming; frequency_value, frequency_unit) to medication patient data criteria window
- added Fulfillments (dispense_datetime, quantity_dispensed_value, quantity_dispensed_unit) to medication patient data criteria
- proposed testing strategy
  - checkout this branch
  - stop any running server(s)
  - update bundle via <code>bundle install</code>
  - clear the local cache via <code>rm -rf tmp/cache/</code>
  - launch Bonnie and navigate to <code>CMS156v2</code>
  - create a new patient with the following criteria
    - first: test
    - last: patient
    - birthdate: 01/01/1900
    - expected values: <code>IPP</code>, <code>DEN</code>
    - Encounter Performed: Office Visit
      - start date: xx/xx/2012
      - end date: xx/xx/2012
    - Medication Order: High-Risk Medications With Days Supply Criteria
      - dose: 5 mg
      - frequency: 1 day(s)
      - fulfillment(s)
        - start date: xx/xx/2012
        - quantity dispensed: < 450 mg for failing, >= 450 mg for passing, across all fulfillments
  - you can tweak the medication dosage, frequency, and fulfillments to test CMD calculations accordingly
